api/src/main/java/org/apache/cloudstack/api/command/user/gui/themes/CreateGuiThemeCmd.java

```diff
@@ -0,0 +1,131 @@
+
+    @Parameter(name = ApiConstants.DESCRIPTION, type = CommandType.STRING, length = 4096, description = "A description for the theme.")
+    private String description;
+
+    @Parameter(name = ApiConstants.CSS, type = CommandType.STRING, length = 65535, description = "The CSS to be retrieved and imported into the GUI " +
+            "when matching the theme access configurations.")
+    private String css;
+
+    @Parameter(name = ApiConstants.JSON_CONFIGURATION, type = CommandType.STRING, length = 65535, description = "The JSON with the configurations to be " +
+            "retrieved and imported into the GUI when matching the theme access configurations.")
+    private String jsonConfiguration;
+
+    @Parameter(name = ApiConstants.COMMON_NAMES, type = CommandType.STRING, length = 65535, description = "A set of Common Names (CN) (fixed or " +
+            "wildcard) separated by comma that can retrieve the theme; e.g.: *acme.com,acme2.com")
+    private String commonNames;
+
+    @Parameter(name = ApiConstants.DOMAIN_IDS, type = CommandType.STRING, length = 65535, description = "A set of domain UUIDs (also known as ID for " +
+            "the end-user) separated by comma that can retrieve the theme.")
+    private String domainIds;
+
+    @Parameter(name = ApiConstants.ACCOUNT_IDS, type = CommandType.STRING, length = 65535, description = "A set of account UUIDs (also known as ID for" +
+            " the end-user) separated by comma that can retrieve the theme.")
+    private String accountIds;
+
+    @Parameter(name = ApiConstants.IS_PUBLIC, type = CommandType.BOOLEAN, description = "Defines whether a theme can be retrieved by anyone when only " +
+            "the `commonNames` is informed. If the `domainIds` or `accountIds` is informed, it is considered as `false`.")
+    private Boolean isPublic = true;
+
+    @Parameter(name = ApiConstants.RECURSIVE_DOMAINS, type = CommandType.BOOLEAN, description = "Defines whether the subdomains of the informed domains are considered. Default " +
+            "value is false.")
+    private Boolean recursiveDomains = false;
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getCss() {
+        return css;
+    }
+
+    public String getJsonConfiguration() {
+        return jsonConfiguration;
+    }
+
+    public String getCommonNames() {
+        return commonNames;
+    }
+
+    public String getDomainIds() {
+        return domainIds;
+    }
+
+    public String getAccountIds() {
+        return accountIds;
+    }
+
+    public Boolean getPublic() {
+        return isPublic;
+    }
+
+    public Boolean getRecursiveDomains() {
+        return recursiveDomains;
+    }
+
+    @Override
+    public void execute() {
+        CallContext.current().setEventDetails(String.format("Name: %s, AccountIDs: %s, DomainIDs: %s, RecursiveDomains: %s, CommonNames: %s", getName(), getAccountIds(),
```

Comment:
i would rather exprect this on the service call and not in the command class. What were your considderations to put it here @BryanMLima ?

Reply:
Yep, I agree, it does make sense to put it there and not in the command class, I will address this.